backend/src/services/logbooks-service.js

```diff
@@ -0,0 +1,232 @@
+            pmhx_dmii: phmxDMII,
+            pmhx_dlp: phmxDLP,
+            pmhx_cva: phmxCVA,
+            exam_weight: examWeight,
+            exam_height: examHeight,
+            exam_bmi: examBMI,
+            exam_veins: examVeins,
+            exam_allen_test: examAllenTest,
+            exam_pulses_sixtytwo: examPulsesSixtyTwo,
+            exam_pulses_sixtyfive: examPulsesSixtyFive,
+            exam_pulses_sixtyeight: examPulsesSixtyEight,
+            exam_pulses_seventy: examPulsesSeventy,
+            cath: cathLink,
+            invx_echo_ef: invxEchoEF,
+            invx_echo_rvfx: invxEchoRVFx,
+            invx_wma: invxWMA,
+            invx_aorta: invxAorta,
+            invx_valves: invxValves,
+            invx_cxr: invxCXR,
+            invx_hb: invxHb,
+            invx_w: invxW,
+            invx_pit: invxPit,
+            invx_cr: invxCr,
+            ct: ctLink,
+            surgical_plan: surgicalPlan,
+            surgical_plan_first_operator: surgicalPlanFirstOperator,
+            surgical_plan_issue_or: surgicalPlanIssueOR,
+            surgical_plan_issue_post: surgicalPlanIssuePost,
+            surgical_plan_flag_for_fu: surgicalPlanaFlagForFU,
+            operative_notes_cpb_h: operativeNotesCPBh,
+            operative_notes_cpb_m: operativeNotesCPBm,
+            operative_notes_xc_h: operativeNotesXCh,
+            operative_notes_xc_m: operativeNotesXCm,
+            operative_notes_ca_h: operativeNotesCAh,
+            operative_notes_ca_m: operativeNotesCAm,
+            my_role: myRole,
+            post_operative_course: postOperativeCourse,
+            learning_points_key_lessons: learningPointsKeyLessons,
+        })
+        .select(); 
+    if (error) {
+        throw new Error(error.message);
+    }
+    return data[0];
+}
+
+async function getLogbookType(logbookID, supabase) {
+    const { data, error } = await supabase.from("logbooks").select().eq("id", logbookID);
+    if (error) {
+        throw new Error(error.message);
+    }
+    return data[0]["type"];
+}
+
+export async function getLogbookLogs(req) {
+    try {
+        const supabase = req.supabase;
+        const { logbookID } = req.params;
+        const logbookType = await getLogbookType(logbookID, supabase);
+        const { data, error } = await supabase.from(logbookType).select();
+        if (error) {
+            throw new Error(error.message);
+        }
+        return data;
+    } catch (error) {
+        throw new Error(error.message);
+    }
+}
+
+export async function getLog(req) {
+    try {
+        const supabase = req.supabase;
+        const { logbookID, logID } = req.params;
+        const logbookType = await getLogbookType(logbookID, supabase);
+        const { data, error } = await supabase.from(logbookType).select().eq("id", logID);
+        if (error) {
+            throw new Error(error.message);
+        }
+        return data[0];
+    } catch (error) {
+        throw new Error(error.message);
```

Comment:
The server will crash if an error is thrown here. Instead in the catch blocks return the error and handle the error in the controller side `routes/logRoutes.js` by returning a 500 status code there. Do the same for any catch blocks in functions that are called by the api routes.